#Entrega 1 - Big Data
###Emilly Mickeli Depine da Silva 23025480
###Renan Teixeira Pinheiro 23025274
###Gustavo Henrique Santos Araujo 23025397
###Fernando José dos Santos 23025299

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import csv

In [ ]:
# Integrando o Colab com o meu Drive para possibilitar a leitura dos arquivos de qualquer dispositivo
caminho_dados = '/content/drive/MyDrive/BaseDadosUber/'

# Lendo os arquivos e definindo o separador padrão como ponto e virgula, para não confundir as virgulas com separadores
product_df = pd.read_csv(caminho_dados + 'product.csv', sep=';')
ride_v2_df = pd.read_csv(caminho_dados + 'ride_v2.csv', sep=';')
rideaddress_v1_df = pd.read_csv(caminho_dados + 'rideaddress_v1.csv', sep=';')
rideestimative_v3_df = pd.read_csv(caminho_dados + 'rideestimative_v3.csv', sep=';')

<ipython-input-7-5de2d7ce2e12>:12: DtypeWarning: Columns (7,8) have mixed types. Specify dtype option on import or set low_memory=False.
  rideaddress_v1_df = pd.read_csv(caminho_dados + 'rideaddress_v1.csv', sep=';')


In [ ]:
# Somando quantas entradas cada coluna tem
print("\nDescrição do product.csv:")
print(product_df.describe().loc[['count']])

print("\nDescrição do ride_v2.csv:")
print(ride_v2_df.describe().loc[['count']])

print("\nDescrição do rideaddress_v1.csv:")
print(rideaddress_v1_df.describe().loc[['count']])

print("\nDescrição do rideestimative_v3.csv:")
print(rideestimative_v3_df.describe().loc[['count']])


Descrição do product.csv:
       ProviderID  CategoryID
count       237.0       237.0

Descrição do ride_v2.csv:
         RideID  RideStatusID  CompanyID  ProviderID     price  CategoryID  \
count  500000.0      500000.0   500000.0    228157.0  500000.0     24714.0   

       TotalUsers  RideDriverLocationID  ScheduledRide  
count    500000.0               14864.0       500000.0  

Descrição do rideaddress_v1.csv:
       RideAddressID  RideAddressTypeID     RideID
count      1000000.0          1000000.0  1000000.0

Descrição do rideestimative_v3.csv:
       RideEstimativeID     RideID  WaitingTime      Price   Selected  \
count         2000000.0  2000000.0    2000000.0  2000000.0  2000000.0   

       RideReasonSelectedEstimativeID        Fee  
count                        234021.0  2000000.0  


In [ ]:
# Informações sobre os tipos de dados e valores não nulos
print("\nInformações sobre o product.csv:")
print(product_df.info())

print("\nInformações sobre o ride_v2.csv:")
print(ride_v2_df.info())

print("\nInformações sobre o rideaddress_v1.csv:")
print(rideaddress_v1_df.info())

print("\nInformações sobre o rideestimative_v3.csv:")
print(rideestimative_v3_df.info())

# Contagem de valores ausentes por coluna
print("\nValores ausentes em product.csv:")
print(product_df.isnull().sum())

print("\nValores ausentes em ride_v2.csv:")
print(ride_v2_df.isnull().sum())

print("\nValores ausentes em rideaddress_v1.csv:")
print(rideaddress_v1_df.isnull().sum())

print("\nValores ausentes em rideestimative_v3.csv:")
print(rideestimative_v3_df.isnull().sum())


Informações sobre o product.csv:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 237 entries, 0 to 236
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   ProductID    237 non-null    object
 1   ProviderID   237 non-null    int64 
 2   CategoryID   237 non-null    int64 
 3   Description  237 non-null    object
dtypes: int64(2), object(2)
memory usage: 7.5+ KB
None

Informações sobre o ride_v2.csv:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500000 entries, 0 to 499999
Data columns (total 15 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   RideID                500000 non-null  int64  
 1   UserID                500000 non-null  object 
 2   Schedule              500000 non-null  object 
 3   Create                500000 non-null  object 
 4   RideStatusID          500000 non-null  int64  
 5   CompanyID             500000 non-null  int64  


In [ ]:
# Verificação de valores duplicados
print("\nValores duplicados em product.csv:")
print(product_df.duplicated().sum())

print("\nValores duplicados em ride_v2.csv:")
print(ride_v2_df.duplicated().sum())

print("\nValores duplicados em rideaddress_v1.csv:")
print(rideaddress_v1_df.duplicated().sum())

print("\nValores duplicados em rideestimative_v3.csv:")
print(rideestimative_v3_df.duplicated().sum())

# Verificação de valores únicos em colunas importantes (ex: id)
print("\nValores únicos em 'product_id' de product.csv:")
print(product_df['ProductID'].nunique())

print("\nValores únicos em 'ride_id' de ride_v2.csv:")
print(ride_v2_df['RideID'].nunique())

# Produto mais usado em product.csv
produto_mais_usado_product = product_df['ProductID'].mode()[0]
contagem_produto_mais_usado_product = product_df['ProductID'].value_counts()[produto_mais_usado_product]

# CompanyID mais usada em ride_v2
company_mais_usada = ride_v2_df['CompanyID'].mode()[0]
contagem_company_mais_usada = ride_v2_df['CompanyID'].value_counts()[company_mais_usada]
print(f"\nA CompanyID mais usada em ride_v2 é '{company_mais_usada}', usada {contagem_company_mais_usada} vezes.")

#Contagem de preços zero em ride_v2
contagem_precos_zero = (ride_v2_df['price'] == 0).sum()
print(f"\nExistem {contagem_precos_zero} corridas com preço zero em ride_v2.")

# Produto mais usado em rideestimative_v3
produto_mais_usado = rideestimative_v3_df['ProductID'].mode()[0]
contagem_produto_mais_usado = rideestimative_v3_df['ProductID'].value_counts()[produto_mais_usado]
print(f"\nO ProductID mais usado em rideestimative_v3 é '{produto_mais_usado}', usado {contagem_produto_mais_usado} vezes.")

#Cidade mais frequente em rideaddress_v1
cidade_mais_frequente = rideaddress_v1_df['City'].mode()[0]
contagem_cidade_mais_frequente = rideaddress_v1_df['City'].value_counts()[cidade_mais_frequente]
print(f"\nA cidade mais frequente em rideaddress_v1 é '{cidade_mais_frequente}', que aparece {contagem_cidade_mais_frequente} vezes.")


Valores duplicados em product.csv:
0

Valores duplicados em ride_v2.csv:
0

Valores duplicados em rideaddress_v1.csv:
0

Valores duplicados em rideestimative_v3.csv:
0

Valores únicos em 'product_id' de product.csv:
236

Valores únicos em 'ride_id' de ride_v2.csv:
500000

A CompanyID mais usada em ride_v2 é '40', usada 144654 vezes.

Existem 26630 corridas com preço zero em ride_v2.

O ProductID mais usado em rideestimative_v3 é 'UberX', usado 235734 vezes.

A cidade mais frequente em rideaddress_v1 é 'São Paulo', que aparece 127724 vezes.


#**Outras conclusões e inferências**

#**Planilha ride_v2.csv**
*   Existem 500.000 registros de corridas.
*   Os RideIDs seguem uma faixa numérica específica, o que é esperado para identificadores únicos.
*  As CompanyIDs variam amplamente (de 1 a 292), sugerindo que várias empresas estão envolvidas.
* A contagem de ProviderID é menor que a contagem total de corridas, o que significa que nem todas as corridas têm um ProviderID associado. Isso pode indicar valores ausentes ou corridas que não se aplicam a um provedor específico.
* Os preços das corridas variam muito (de 0 a 15254.80), com um desvio padrão alto.
* Existem 26630 corridas com preço 0, o que sugere a quantidade de corridas canceladas ou reembolsadas
* A maioria das corridas tem apenas um usuário (a mediana é 1), mas existem corridas com até 4 usuários
* A grande maioria das corridas não é agendada (a mediana é 0)


##**Planilha rideaddress_v1.csv:**
* Existem 1.000.000 registros de endereços de corrida.
* RideAddressTypeID varia entre 1 e 2, provavelmente indicando tipos de endereços (origem e destino, por exemplo).
* Os RideID dessa tabela, correspondem aos RideID da tabela ride_v2.
* Os Id's de ambas as colunas Id, seguem uma distribuição uniforme.

##**Planilha rideestimative_v3.csv**
* Existem 2.000.000 estimativas de corrida.
* Os RideID dessa tabela, correspondem aos RideID da tabela ride_v2
* O tempo de espera médio é de aproximadamente 6 minutos, com uma variação relativamente pequena.